pageserver/src/tenant/timeline.rs

```diff
@@ -2321,13 +2326,21 @@ impl Timeline {
     }
 
     pub(crate) fn get_lazy_slru_download(&self) -> bool {
+        if self.lazy_slru_download.load(AtomicOrdering::Relaxed) {
+            return true;
+        }
```

Comment:
Caching this after the first basebackup leads to strange load pattern on the pageserver.

I think it would be better to check the SLRU size before serving the first basebackup.
You can do that by getting the size of `Timeline::get_slru_keyspace`. It reads the slru segment directory for each slru kind and then one read per segment to get the size, so should be fairly cheap. Might even be cheap enough that we can do it every time without caching.